autoscale/src/main/java/com/sequenceiq/periscope/monitor/CleanupMonitor.java

```diff
@@ -0,0 +1,48 @@
+package com.sequenceiq.periscope.monitor;
+
+import static com.sequenceiq.periscope.api.model.ActivityStatus.SCALING_FLOW_FAILED;
+import static com.sequenceiq.periscope.api.model.ActivityStatus.SCALING_FLOW_SUCCESS;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.periscope.api.model.ActivityStatus;
+import com.sequenceiq.periscope.model.ScalingActivities;
+import com.sequenceiq.periscope.monitor.evaluator.EvaluatorExecutor;
+import com.sequenceiq.periscope.monitor.evaluator.cleanup.ActivityCleanupEvaluator;
+
+@Component
+@ConditionalOnProperty(prefix = "periscope.enabledAutoscaleMonitors.cleanup-monitor", name = "enabled", havingValue = "true")
+public class CleanupMonitor extends ScalingActivityMonitor {
+
+    @Override
+    protected List<ScalingActivities> getMonitored() {
+        Set<ActivityStatus> statuses = EnumSet.of(SCALING_FLOW_FAILED, SCALING_FLOW_SUCCESS);
```

Comment:
Why is this limited to just these 2 states ? Won't that end up leaving a bunch of the other events in the table, which then causes it to grow indefinitely.

Reply:
I initially used these two because they are final states. Updated this to consider additional states - `METRICS_COLLECTION_FAILED`, `UPSCALE_TRIGGER_FAILED`, `DOWNSCALE_TRIGGER_FAILED`. Do we need to account for any states apart from these?